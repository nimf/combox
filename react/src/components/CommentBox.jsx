import React from 'react';
import PropTypes from 'prop-types';
import { Form, Button } from 'semantic-ui-react';

const propTypes = {
  text: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  enabled: PropTypes.bool.isRequired,
};
const defaultProps = {
  text: '',
};

function CommentBox({
  text, onChange, onSubmit, enabled,
}) {
  return (
    <Form onSubmit={onSubmit}>
      <Form.TextArea
        autoHeight
        rows={2}
        value={text}
        onChange={onChange}
        placeholder="Leave a comment..."
      />
      <Button
        disabled={!enabled || text.trim().length === 0}
        content="Add Comment"
        labelPosition="left"
        icon="edit"
        primary
      />
    </Form>
  );
}

CommentBox.propTypes = propTypes;
CommentBox.defaultProps = defaultProps;

export default CommentBox;
