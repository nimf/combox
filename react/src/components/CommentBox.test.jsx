import React from 'react';
import renderer from 'react-test-renderer';
import CommentBox from './CommentBox';

describe('CommentBox renders', () => {
  test('renders with text with enabled button', () => {
    const tree = renderer.create(<CommentBox
      text="Test comment"
      enabled={true}
      onChange={() => {}}
      onSubmit={() => {}}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders without text with disabled button', () => {
    const tree = renderer.create(<CommentBox
      text=""
      enabled={true}
      onChange={() => {}}
      onSubmit={() => {}}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders disabled with disabled button', () => {
    const tree = renderer.create(<CommentBox
      text="Test comment"
      enabled={false}
      onChange={() => {}}
      onSubmit={() => {}}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
