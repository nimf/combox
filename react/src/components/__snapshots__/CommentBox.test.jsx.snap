// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox renders renders disabled with disabled button 1`] = `
<form
  action={undefined}
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <textarea
      content={undefined}
      disabled={undefined}
      onChange={[Function]}
      onInput={[Function]}
      placeholder="Leave a comment..."
      required={undefined}
      rows={2}
      style={
        Object {
          "resize": "none",
        }
      }
      type={undefined}
      value="Test comment"
    />
  </div>
  <button
    className="ui icon primary disabled left labeled button"
    disabled={true}
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <i
      aria-hidden="true"
      className="edit icon"
    />
    Add Comment
  </button>
</form>
`;

exports[`CommentBox renders renders with text with enabled button 1`] = `
<form
  action={undefined}
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <textarea
      content={undefined}
      disabled={undefined}
      onChange={[Function]}
      onInput={[Function]}
      placeholder="Leave a comment..."
      required={undefined}
      rows={2}
      style={
        Object {
          "resize": "none",
        }
      }
      type={undefined}
      value="Test comment"
    />
  </div>
  <button
    className="ui icon primary left labeled button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    <i
      aria-hidden="true"
      className="edit icon"
    />
    Add Comment
  </button>
</form>
`;

exports[`CommentBox renders renders without text with disabled button 1`] = `
<form
  action={undefined}
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <textarea
      content={undefined}
      disabled={undefined}
      onChange={[Function]}
      onInput={[Function]}
      placeholder="Leave a comment..."
      required={undefined}
      rows={2}
      style={
        Object {
          "resize": "none",
        }
      }
      type={undefined}
      value=""
    />
  </div>
  <button
    className="ui icon primary disabled left labeled button"
    disabled={true}
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <i
      aria-hidden="true"
      className="edit icon"
    />
    Add Comment
  </button>
</form>
`;
